src/features/favorites/pages/reducer.ts

```diff
@@ -1,26 +1,34 @@
-import { FavoriteResponse } from 'api/gen'
+import { FILTER_OPTIONS } from 'features/favorites/pages/Filters'
 
 export type FavoritesParameters = {
-  favorites: Array<FavoriteResponse>
+  filter: keyof typeof FILTER_OPTIONS | undefined
 }
 
 export type FavoritesState = FavoritesParameters & {
   showResults: boolean
+  filter: keyof typeof FILTER_OPTIONS | undefined
   dispatch: (action: Action) => void
 }
 
 export const initialFavoritesState: FavoritesState = {
   showResults: false,
-  favorites: [],
+  filter: undefined,
   dispatch: () => null,
 }
 
-export type Action = { type: 'SHOW_RESULTS'; payload: boolean }
+export type Action =
+  | { type: 'INIT' }
+  | { type: 'SHOW_RESULTS'; payload: boolean }
+  | { type: 'SET_FILTER'; payload: keyof typeof FILTER_OPTIONS | undefined }
 
 export const favoritesReducer = (state: FavoritesState, action: Action): FavoritesState => {
   switch (action.type) {
+    case 'INIT':
+      return { ...initialFavoritesState, dispatch: state.dispatch, showResults: state.showResults }
     case 'SHOW_RESULTS':
       return { ...state, showResults: action.payload }
+    case 'SET_FILTER':
+      return { ...state, filter: action.payload }
```

Comment:
je suis contre les action.payload... Je ne sais même si cela a eu du sens un jour
```suggestion
      return { ...state, filter: action.filters }
```

Reply:
100% d'accord.

Reply:
comme vous voulez, mais il faut rester consistant et changer le reducer de la recherche dans ce cas là

Reply:
Je propose de garder la similitude entre les deux et faire cela dans un prochain PR refactor search/favorites.